tensorflow/lite/micro/kernels/cmsis-nn/pooling.cc

```diff
@@ -387,7 +352,7 @@ TfLiteRegistration* Register_AVERAGE_POOL_2D() {
 TfLiteRegistration* Register_MAX_POOL_2D() {
   static TfLiteRegistration r = {/*init=*/pooling::Init,
                                  /*free=*/nullptr,
-                                 /*prepare=*/pooling::Prepare,
```

Comment:
Why set Prepare to nullptr?  Aren't you doing valuable checks?

Reply:
well, the prepare() was just allocating a scratch buffer irrespective of the pooling type. And max pool doesn't require additional buffers.

I've split the Prepare as AveragePrepare() and MaxPrepare() , with checks , CalcOpData() and scratch buffer allocation only for the right pooling type.

Reply:
I see, this looks much better, thanks.